transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -286,23 +329,21 @@ protected void run() {
                     default:
                         // fallthrough
                 }
-
+                lastBlockingAmount.lazySet(System.nanoTime() - runStartTime);
                 final int ioRatio = this.ioRatio;
+                final long ioStartTime = System.nanoTime();
+                lastProcessedChannels.lazySet(tryProcessReady(events, strategy));
+                final long ioEndTime = System.nanoTime();
+                final long ioTime = ioEndTime - ioStartTime;
                 if (ioRatio == 100) {
-                    if (strategy > 0) {
-                        processReady(events, strategy);
-                    }
                     runAllTasks();
                 } else {
-                    final long ioStartTime = System.nanoTime();
-
-                    if (strategy > 0) {
-                        processReady(events, strategy);
-                    }
-
-                    final long ioTime = System.nanoTime() - ioStartTime;
                     runAllTasks(ioTime * (100 - ioRatio) / ioRatio);
                 }
+                final long taskTime = System.nanoTime() - ioTime;
+                lastIoProcessingTime.lazySet(ioTime * 1000);
```

Comment:
> My suggestion would be to not support multiple listeners (if that was a point of discussion).

Agreed. This can always be implemented in the single listener if the user needs it.